pkg/spec/spec.go

```diff
@@ -309,31 +337,11 @@ func CreateConfigToOCISpec(config *CreateConfig) (*spec.Spec, error) { //nolint
 	if err != nil {
 		return nil, errors.Wrapf(err, "error getting volume mounts")
 	}
-	if len(mounts) > 0 {
-		// If we have overlappings mounts, remove them from the spec in favor of
-		// the user-added volume mounts
-		destinations := make(map[string]bool)
-		for _, mount := range mounts {
-			destinations[path.Clean(mount.Destination)] = true
-		}
 
-		// Copy all mounts from spec to defaultMounts, except for
-		//  - mounts overridden by a user supplied mount;
-		//  - all mounts under /dev if a user supplied /dev is present;
-		mountDev := destinations["/dev"]
-		for _, mount := range configSpec.Mounts {
-			if _, ok := destinations[path.Clean(mount.Destination)]; !ok {
-				if mountDev && strings.HasPrefix(mount.Destination, "/dev/") {
-					// filter out everything under /dev if /dev is user-mounted
-					continue
-				}
-
-				logrus.Debugf("Adding mount %s", mount.Destination)
-				mounts = append(mounts, mount)
-			}
-		}
-		configSpec.Mounts = mounts
-	}
+	configSpec.Mounts = supercedeUserMounts(mounts, configSpec.Mounts)
+	//--mount
+	fsMounts := config.GetFSMounts(configSpec.Mounts)
```

Comment:
I don't really know that this is necessary - do we really need to go back through all the mounts we've just added and rework their options?

Reply:
Yes I am not sure what he was doing here dropping.